doc/conceptualOverview.dox

```diff
@@ -57,30 +56,161 @@ Poll inputs do not have the AccessMode::wait_for_new_data flag, and hence all re
+
+\snippet{lineno} example/src/AverageCurrent.cc mainLoop implementation
 
-\section conceptualOverview_ModuleGroup Module groups
 
 \section conceptualOverview_DeviceModule Device modules
 
-\subsection conceptualOverview_DeviceModule_trigger Triggers
+To connect to a device, a DeviceModule needs to be instantiated in the Application, similar to instantiating an ApplicationModule:
 
-\section conceptualOverview_ControlSystemModule Control system module
+\snippet{lineno} example/include/ExampleApp.h Device
 
-\section conceptualOverview_Application The Application
+This will publish all registers from the catalogue of the specified device "oven" to the control system and to the application modules. Since "oven" is a device alias and not a CDD, we also need to specify the DMAP file by creating an instance of SetDMapFilePath *before* the DeviceModule instance like this:
+
+\snippet{lineno} example/include/ExampleApp.h SetDMapFilePath
+
+The call to getName() returns the name of the Application, which will be provided later to the Application constructor. This allows us to influence the file name for automated tests. The DMAP file in our exmaple speficies two devies:
 
-\section conceptualOverview_Connections Process variable connections
+\include example/config/demo_example.dmap
+
+It is a good practise to use a <a href="https://chimeratk.github.io/DeviceAccess/master/lmap.html">logical name mapping device</a> to allow more control over the representation of the device. The second device is the actaul hardware device which will be used as a target for the logical name mapping device. We are using a sharedMemoryDummy backend (map file \ref example_config_demo_example_dmap "DemoDummy.map"), so we can run the application locally and interact with the dummy device through QtHardMon.
+
+The \ref example_config_oven_xlmap "example xlmap mapping file" generates the following variable hierarchy:
+
+- /Configuration/heaterMode (redirects to to HEATER.MODE, read/write)
+- /Configuration/lightOn (redirects to BOARD.GPIO_OUT0, read/write)
+- /Controller/heatingCurrent (reditects to HEATER.CURRENT_SET, read/write)
+- /Controller/temperatureReadback (redirects to SENSORS.TEMPERATURE1, read only)
+- /Monitoring/heatingCurrent (redirects to HEATER.CURRENT_READBACK, read only)
+- /Monitoring/temperatureOvenTop (redirects to SENSORS.TEMPERATURE2, read only)
+- /Monitoring/temperatureOvenBottom (redirects to SENSORS.TEMPERATURE3, read only)
+- /Monitoring/temperatureOutside (redirects to SENSORS.TEMPERATURE4, read only)
+
+Since in ApplicationCore, all device registers are treated as unidirectional, read/write registers will be used in the write direction only. Hence, the first 3 variables will have the device as a consumer, receiving values either from an ApplicationModule or from the control system. The other variables will use the device as a feeder, so the device will provide values to the ApplicationModules and the control system.
```

Comment:
Wrong comma behind `Since in ApplicationCore`. 
```suggestion
In ApplicationCore  all device registers are treated as unidirectional, and read/write registers will be used in the write direction only. Hence, the first 3 variables will have the device as a consumer, receiving values either from an ApplicationModule or from the control system. The other variables will use the device as a feeder, so the device will provide values to the ApplicationModules and the control system.
```

Reply:
Side remark: Lines are very long. There should be line breaks after each full stop.

Reply:
We agreed on not making any line breaks to format the Doxygen source code when writing the specs (for very good reasons, because that creates a total mess). Use an editor with dynamic line breaks :-)

Reply:
In the review it always gives you large chunks to refer to. For editing it is fine.

Reply:
I still think it is the lesser of the two issues. Reviews with line breaks were also quite messy, because of the additional line breaks introduced by Github.